### DISCUSSION TITLE
Starting and ending values

### DISCUSSION THREAD

**Discussion by MaksBan** [patrykandpatrick/vico#594] [Questions]
Hey, I want to achieve effect like this: graph is starting with 0m and ends with whatever the last value is.

![image](https://github.com/patrykandpatrick/vico/assets/81077310/b9cdc5e3-9f11-4b33-94be-07c3ca45921e)

Right now my graph looks like this:

![image](https://github.com/patrykandpatrick/vico/assets/81077310/32fc42e5-0686-4016-8e7a-f2642ba63266)

If anybody here know how to resolve this? Thanks! :D

**Comment by patrickmichalik** [comment]
Hello! Are you using Vico 1 or Vico 2, and are you using it with Jetpack Compose or the view system?

**Reply by MaksBan** [reply]
Hi! I forgot about this informations, sorry.

Im using Jetpack Compose with kotlin 1.9.21 and vicoVersion = "1.11.1".

**Reply by patrickmichalik** [reply]
Got it. The functionality in question was introduced in Vico 1.13.0. (I’d suggest updating to Vico 1.14.0 or Vico 2.0.0 Alpha 12.) `AxisItemPlacer.Horizontal.default` now has an `addExtremeLabelPadding` parameter, and when `true` is passed in, padding is added for the first and last `HorizontalAxis` labels. You can make the whole `LineChart` line visible by turning off scroll—see the `isScrollEnabled` parameter of `rememberChartScrollSpec`. Alignment customization for `HorizontalAxis` labels is a planned addition.

**Comment by claude** [comment]
Following up here, since the alignment customization mentioned above has now shipped. Vico [3.3.0](https://github.com/patrykandpatrick/vico/releases/tag/v3.3.0) lets you label the first and last _x_ values without the plot area shrinking to accommodate them:

- `HorizontalAxis.ItemPlacer.extremes` places labels, ticks, and guidelines only at the first and last _x_ values, with the first label end-aligned and the last start-aligned, so both sit flush with the plot-area edges.
- `HorizontalAxis.ItemPlacer.aligned` gains a `shiftExtremeLabels` parameter, which applies the same inward alignment to the outermost labels while keeping regular spacing in between.

```kotlin
bottomAxis =
  HorizontalAxis.rememberBottom(
    itemPlacer = remember { HorizontalAxis.ItemPlacer.extremes() }
  )
```

This is the alternative to `addExtremeLabelPadding` that was planned back then: no `CartesianLayer` padding is added, so the line still spans the full width of the chart.

<sub>Based on maintainer input.</sub>
